src/main/java/lotto/io/UserLottoInput.java

```diff
@@ -0,0 +1,71 @@
+package lotto.io;
+
+import camp.nextstep.edu.missionutils.Console;
+import java.util.Arrays;
+import java.util.List;
+import lotto.domain.Lotto;
+import lotto.domain.LottoBall;
+import lotto.exception.InvalidRangeInputException;
+import lotto.exception.InvalidInputException;
+
+public class UserLottoInput implements LottoInput {
+    private final String DELIMITER = ",";
+
+    @Override
+    public int getMoneyAmount() {
+        String input = Console.readLine();
+        validateMoneyAmount(input);
+        return Integer.parseInt(input);
+    }
+
+    private void validateMoneyAmount(String input) {
+        validateInteger(input);
+        validatePositive(input);
+    }
+
+    private void validatePositive(String input) {
+        if (Integer.parseInt(input) <= 0) {
+            throw new InvalidRangeInputException();
+        }
+    }
+
+    @Override
+    public Lotto getLotto() {
+        List<LottoBall> winningLottoBalls = getNumbers()
+                .stream()
+                .map(LottoBall::new)
+                .toList();
+
+        return new Lotto(winningLottoBalls);
+    }
+
+    private List<Integer> getNumbers() {
+        String input = Console.readLine();
+        String[] numbersInput = input.split(DELIMITER);
+
+        Arrays.stream(numbersInput)
+                .forEach(this::validateInteger);
+
+        return Arrays.stream(numbersInput)
+                .map(Integer::parseInt)
+                .toList();
+    }
```

Comment:
```suggestion
    private List<Integer> getNumbers() {
        String input = Console.readLine();
        String[] numbersInput = input.split(DELIMITER);

        return Arrays.stream(numbersInput)
                .map(this::parseInt)
                .toList();
    }
    
    private Integer parseInt(String input) {
        try {
            return Integer.parseInt(input);
        } catch (NumberFormatException e) {
            throw new InvalidInputException();
        }
    }
```
기존 코드에서는 파싱하며 검증한 다음 다시 파싱해서 중복되니, 이러면 어떨까요?

Reply:
기존 메서드를 래핑해서 사용하면 예외를 섬세하게 다룰 수 있네요. 좋은 제안인 것 같습니다!